src/main/scala/firrtl/options/Stage.scala

```diff
@@ -45,8 +46,14 @@ abstract class Stage extends Phase {
     * @param initialAnnotations annotation
     * @return output annotations
     */
-  final def execute(args: Array[String], annotations: AnnotationSeq): AnnotationSeq =
+  final def execute(args: Array[String], annotations: AnnotationSeq): AnnotationSeq = try {
     transform(shell.parse(args, annotations))
+  } catch {
+    case OptionsHelpException => annotations
+    case OptionsException(msg, _) =>
+      StageUtils.dramaticUsageError(msg)
+      annotations
+  }
```

Comment:
Why is this continuing and not fully erroring?

Reply:
This is now updated to call `System.exit(1)` here. This should clean up all of the issues with the FIRRTL fat jar not returning normal exit codes. Tests are added to make sure that the exit codes behave as expected.